frontend/src/components/worksheets/Worksheet.js

```diff
@@ -511,9 +514,17 @@ class Worksheet extends React.Component {
             Mousetrap.bind(
                 ['shift+r'],
                 function(e) {
-                    // refresh the worksheet and update the
-                    // focus to the first item
-                    this.reloadWorksheet(undefined, 'end');
+                    var focusIndex = this.state.focusIndex;
+                    var subFocusIndex = this.state.subFocusIndex;
+                    this.reloadWorksheet(focusIndex, subFocusIndex);
```

Comment:
Looking at the reloadworksheet, this doesn't seem the correct arguments (passing focusIndex as partialUpdateItems?).  

I think a better way might be to add a new argument (similar to rawIndexAfterEditMode) for refresh, after refetching the worksheet, setFocus on the original focus&subfocus. We actually want to refetch the worksheet after shift+r I think, so partialUpdateItem should be undefined for this call